File: tailwind-admin-reactjs-free/package/src/types/stats.ts
export interface ConteoItem {
  etiqueta: string;
  cantidad: number;
}

export interface AsistenciaResumen {
  total_registros: number;
  presentes: number;
  porcentaje_presente: number | null;
}

export interface DashboardStats {
  estudiantes_activos: number;
  profesores_activos: number;
  encargados_activos: number;
  grupos: number;
  materias: number;
  subgrupos: number;
  evaluaciones: number;
  promedio_general_notas: number | null;
  asistencia: AsistenciaResumen;
  distribucion_asistencia: ConteoItem[];
  estudiantes_por_grupo: ConteoItem[];
}

File: backend/app/schemas/stats.py
from pydantic import BaseModel


class ConteoItem(BaseModel):
    etiqueta: str
    cantidad: int


class AsistenciaResumen(BaseModel):
    total_registros: int
    presentes: int
    porcentaje_presente: float | None


class DashboardStats(BaseModel):
    estudiantes_activos: int
    profesores_activos: int
    encargados_activos: int
    grupos: int
    materias: int
    subgrupos: int
    evaluaciones: int
    promedio_general_notas: float | None
    asistencia: AsistenciaResumen
    distribucion_asistencia: list[ConteoItem]
    estudiantes_por_grupo: list[ConteoItem]
